aiohomekit/controller/ble/pairing.py

```diff
@@ -409,6 +421,59 @@ async def _process_disconnected_events(self) -> None:
         for listener in self.listeners:
             listener(results)
 
+    def _async_notification(self, data: HomeKitEncryptedNotification) -> None:
+        """Receive a notification from the accessory."""
+        logger.warning("%s: Received notification: %s", self.name, data)
+
+    async def _async_set_broadcast_encryption_key(self) -> None:
+        """Get the broadcast key for the accessory."""
+        info = self.accessories.aid(1).services.first(service_type=SIGNATURE_SERVICE)
+        hap_char = info[SIGNATURE_SERVICE_CHAR]
+
+        # ble_char = self.client.get_characteristic_by_handle(hap_char.handle)
+        # iid = await self.client.get_characteristic_iid(ble_char)
+        # if iid is None:
+        #    logger.debug("%s: No iid for %s", self.name, hap_char.uuid)
+        #    return
+        payload = b"\x01\x00"
+
+        service_iid = hap_char.service.iid
+        # service_iid = 7
+        logger.debug(
+            "%s: Setting broadcast key for service_iid: %s",
+            self.name,
+            service_iid,
+        )
+        try:
+            data = await self._async_request_under_lock(
+                OpCode.PROTOCOL_CONFIG, hap_char, payload, iid=service_iid
```

Comment:
This always fails saying its the wrong IID

Reply:
need to make sure we are decoding the service iids correctly

Reply:
Shouldn't it be the char iid?

Reply:
I mean i'm guessing from the `for iid in range(64)` that that doesn't work either 😆

Reply:
It accepts '7' but gives me nothing back

Reply:
> Shouldn't it be the char iid?

The docs say its the service iid and the ADK seems to support that so I'm hoping we are decoding the iid for services wrong somewhere https://github.com/apple/HomeKitADK/blob/fb201f98f5fdc7fef6a455054f08b59cca5d1ec8/HAP/HAPBLEProcedure.c#L213

Reply:
Aaahhhhh of course.

Reply:
I think I fixed it in the last commit.

trying again now

Reply:
7 is the correct iid

`2022-10-21 10:32:13.714 WARNING (MainThread) [aiohomekit.controller.ble.pairing] VOCOlinc-VS1-185620 [6ABB4791-47FB-B455-0543-FEA1F0976DEF] (id=E4:2F:25:E5:71:C5): Got broadcast key for iid: 7: b''`

Reply:
I think the docs might be wrong about how to generate the key. comparing with the ADK

Reply:
The docs say it responds with the key but I don't think it actually does.  I think its just to sync generation.  I think we might have enough data to generate ourselves

Reply:
This one is going to be quite a bit of trial and error as the docs are definitely wrong